core/templates/pages/learner-dashboard-page/goals-tab.component.html

```diff
@@ -1,106 +1,120 @@
+            </p>
+          </a>
+          <img [src]="bookImageUrl" alt="book image" class="oppia-svg-book-image" *ngIf="editGoalsTopicClassification[i] == 1" ngbTooltip="You have completed this topic.">
+          <mat-checkbox color="primary" *ngIf="editGoalsTopicClassification[i] == 2"
+                        class="one oppia-checkbox e2e-test-add-topic-to-current-goals-button"
+                        (click)="this.currentGoals.length !== this.MAX_CURRENT_GOALS_LENGTH && addToLearnerGoals(topic, topic.id, i)"
+                        [disabled]="this.currentGoals.length === this.MAX_CURRENT_GOALS_LENGTH">
+          </mat-checkbox>
+          <mat-checkbox color="primary" *ngIf="editGoalsTopicClassification[i] == 0"  checked class="oppia-checkbox"
+                        (click)="addToLearnerGoals(topic, topic.id, i)">
+          </mat-checkbox>
         </div>
       </div>
     </div>
+    <div tabindex="0" class="oppia-goals-box e2e-test-completed-goals-section">
+      <p class="oppia-goals-title"
+        [innerHTML]="'I18N_LEARNER_DASHBOARD_COMPLETED_GOALS_SECTION' | translate">
+      </p>
+      <div *ngIf="completedGoals.length > 0">
+        <div *ngFor="let topic of completedGoals; let i = index">
+          <div class="oppia-completed-goals-content">
+            <img [src]="starImageUrl" alt="star image" class="oppia-svg-star-image">
+            <a [href]="completedGoalsTopicPageUrl[i]"
+               class="oppia-completed-goals-topic-name e2e-test-completed-goals-topic-name">
+              <span [innerHTML]="'I18N_LEARNT_TOPIC' | translate: { topicName: topic.name }"></span>
+            </a>
+            <div *ngFor="let story of topic.storyTitles">
+              <p class="oppia-goals-story-name">
+                <span [innerHTML]="'I18N_COMPLETED_STORY' | translate: { story: story }"></span>
+              </p>
+            </div>
+          </div>
+        </div>
+      </div>
+      <div *ngIf="completedGoals.length == 0" class="oppia-empty-goals-content">
+        <p [innerHTML]="'I18N_LEARNER_DASHBOARD_EMPTY_COMPLETED_GOALS_SECTION' | translate"></p>
+      </div>
+    </div>
   </div>
-  <div *ngIf="completedGoals.length == 0" class="oppia-empty-goals-content">
-    <p [innerHTML]="'I18N_LEARNER_DASHBOARD_EMPTY_COMPLETED_GOALS_SECTION' | translate"></p>
+  <div *ngIf="learnerDashboardRedesignFeatureFlag">
+    <div class="oppia-learner-dash-section d-flex justify-content-between">
+      <p class="oppia-learner-dash-goals-text mb-0"> {{ 'I18N_LEARNER_DASHBOARD_GOALS_SECTION_INSTRUCTIONS' | translate }} </p>
+      <!--TODO(#18384): Check to see if text is part of button (text changes when hover too)-->
+      <button class="oppia-learner-dash-button--add-goals align-items-center flex-grow-1 d-flex justify-content-center bg-white border-0" (click)="openModal()">
```

Comment:
```suggestion
      <button class="oppia-learner-dash-button-add-goals align-items-center flex-grow-1 d-flex justify-content-center bg-white border-0" (click)="openModal()">
```

Reply:
The class is called .oppia-learner-dash-button--add-goals, the double dash indicates variant of button. I don't think there's another button class I need to create for the goals tab though; should I rename it to .oppia-learner-dash-button--goals instead?

Reply:
Sorry I don't understand "variant of button", can you please elaborate?

This is a trivial comment, please do whatever is more aligned to the project. Thanks.